### PR TITLE
ORLP-172 url hateoas use self link for managing deck

### PR DESCRIPTION
**Pull request by Tamara20**
I use EditDeckDTO and  DeckEditCategoryDTO for creating right DTO, which can be transformed to Deck with property category in server. I save url - self link and use it in managing deck. Please see ORLP-173 in orlp-server before.

### CODE REVIEW

File: src/app/dto/DeckDTO/deck.added.admin.DTO.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {DeckEditCategoryDTO} from './deck.edit.category.DTO';
+
+export class DeckAddedAdminDTO {
```
Comment (b0hdan) [inline src/app/dto/DeckDTO/deck.added.admin.DTO.ts]: This DTO is useless. Use EditDeckDTO instead.

File: src/app/page/admin/decks/admin.decks.component.ts [hunk 2]
```diff
@@ -20,9 +19,11 @@ export class AdminDecksComponent implements OnInit {
   public deckSelected: AdminDeck;
   public categories: CategoryLink[];
   public deckList: AdminDeck[];
-  public chosenCategoryId: number;
+  public categoryForDeckAdd: DeckEditCategoryDTO = new DeckEditCategoryDTO();
+  public categoryForDeck: DeckEditCategoryDTO = new DeckEditCategoryDTO();
```
Comment (b0hdan) [inline src/app/page/admin/decks/admin.decks.component.ts]: One object will be enough.

Comment (Tamara20) [reply src/app/page/admin/decks/admin.decks.component.ts]: yes, I'll fix it

File: src/app/page/admin/decks/admin.decks.service.ts [hunk 3]
```diff
@@ -23,30 +26,38 @@ export class AdminDecksService {
         DTOConverter.jsonArrayToCollection(DTOConverter.jsonToAdminDeck, response.json()))
       .catch(this.handleError);
   }
-
-  deleteDeck(deckId: number) {
-    return this.orlp.delete('/api/admin/decks/' + deckId);
-}
+  deleteDeck(link: Link) {
+    let shortLink: string = this.orlp.getShortLink(link);
+    shortLink = this.orlp.decodeLink(shortLink);
```
Comment (antonbakalets) [inline src/app/page/admin/decks/admin.decks.service.ts]: It feels like you are initializing one variable twice, I would either combine those two lines in one, either use another one on the second line.

File: src/app/page/admin/decks/admin.decks.service.ts [hunk 4]
```diff
@@ -23,30 +26,38 @@ export class AdminDecksService {
...
+  createDeckAdmin (deckAdded: DeckAddedAdminDTO ) {
+    return this.orlp.post('/api/admin/decks', deckAdded)
       .map((response: Response) => response.json());
   }
 
-  editDeck (deckEdit: NewDeckDTO, deck_id: number ) {
-    return this.orlp.put('/api/admin/decks/' + deck_id + '/' + deckEdit.categoryId, deckEdit)
+  editDeck (link: Link, deckEdit: EditDeckDTO) {
+    let shortLink: string = this.orlp.getShortLink(link);
+    shortLink = this.orlp.decodeLink(shortLink);
```
Comment (antonbakalets) [inline src/app/page/admin/decks/admin.decks.service.ts]: The same here.

File: src/app/page/admin/decks/admin.decks.service.ts (hunk 3 above)
Comment (Tamara20) [reply src/app/page/admin/decks/admin.decks.service.ts]: yes it's true, thank you.

**Pull request merged**